packages/next-swc/crates/next-core/src/next_server/context.rs

```diff
@@ -103,142 +103,63 @@ pub async fn get_server_resolve_options_context(
-                plugins: vec![
-                    Vc::upcast(module_feature_report_resolve_plugin),
-                    Vc::upcast(server_component_externals_plugin),
-                    Vc::upcast(unsupported_modules_resolve_plugin),
-                ],
-                ..Default::default()
-            };
-            ResolveOptionsContext {
-                enable_typescript: true,
-                enable_react: true,
-                rules: vec![(
-                    foreign_code_context_condition,
-                    resolve_options_context.clone().cell(),
-                )],
-                ..resolve_options_context
-            }
+        ServerContextType::Pages { .. }
+        | ServerContextType::PagesData { .. }
+        | ServerContextType::AppSSR { .. } => {}
+    };
+    let external_cjs_modules_plugin = ExternalCjsModulesResolvePlugin::new(
+        project_path,
+        ExternalPredicate::AllExcept(next_config.transpile_packages()).cell(),
+    );
+
+    let plugins = match ty {
+        ServerContextType::Pages { .. } | ServerContextType::PagesData { .. } => {
+            vec![
+                Vc::upcast(module_feature_report_resolve_plugin),
+                Vc::upcast(external_cjs_modules_plugin),
+                Vc::upcast(unsupported_modules_resolve_plugin),
+            ]
         }
-        ServerContextType::Middleware => {
-            let resolve_options_context = ResolveOptionsContext {
-                enable_node_modules: Some(root_dir),
-                enable_node_externals: true,
-                module: true,
-                custom_conditions: vec![mode.node_env().to_string()],
-                plugins: vec![
-                    Vc::upcast(module_feature_report_resolve_plugin),
-                    Vc::upcast(unsupported_modules_resolve_plugin),
-                ],
-                ..Default::default()
-            };
-            ResolveOptionsContext {
-                enable_typescript: true,
-                enable_react: true,
-                rules: vec![(
-                    foreign_code_context_condition,
-                    resolve_options_context.clone().cell(),
-                )],
-                ..resolve_options_context
-            }
+        ServerContextType::AppSSR { .. }
+        | ServerContextType::AppRSC { .. }
+        | ServerContextType::AppRoute { .. }
+        | ServerContextType::Middleware { .. } => {
+            vec![
+                Vc::upcast(module_feature_report_resolve_plugin),
+                Vc::upcast(server_component_externals_plugin),
+                Vc::upcast(unsupported_modules_resolve_plugin),
+            ]
         }
+    };
+    let resolve_options_context = ResolveOptionsContext {
+        enable_node_modules: Some(root_dir),
+        enable_node_externals: true,
+        enable_node_native_modules: true,
+        module: true,
+        custom_conditions,
+        import_map: Some(next_server_import_map),
```

Comment:
`ServerContextType::Middleware` didn't used to get the import map, could this be a problem?